### PR TITLE
Add disambiguate tool to QC portion of the weave pipeline

### PR DESCRIPTION
**Pull request by rroutsong**
Partially addresses #25 

This pull request implements disambiguate when a command line user identifies a valid host genome using the --host[-t] and pathogen genome by the --pathogen[-p] flags. If does not use a valid fasta, or gzipped fasta, genome an error will be raised.

Currently this implementation of disambiguate only uses `bwa mem` for alignment. This will need to change in the future including a method of detecting whether a cohort is DNA [bwa mem or support bowtie2] or RNA [STAR-aligner].

### CODE REVIEW

File: docker/NGS_QC/Dockerfile [hunk 1]
```diff
@@ -3,10 +3,13 @@ ENV TZ=America/New_York
...
 RUN apt-get update; apt-get upgrade -y bash && \
-    apt-get install -y fastqc multiqc fastp kraken2 bowtie2 libgd-graph-perl git curl libz-dev build-essential
+    apt-get install -y fastqc python-is-python3 rna-star bwa hisat2 python3-pip samtools multiqc fastp kraken2 bowtie2 libgd-graph-perl git curl libz-dev build-essential
 RUN mkdir /opt2; curl -S -s -L https://github.com/StevenWingett/FastQ-Screen/archive/refs/tags/v0.15.3.tar.gz --output /opt2/v0.15.3.tar.gz
 RUN cd /opt2 && tar xvf v0.15.3.tar.gz 
 RUN cd /opt2 && git clone https://github.com/bioinformatics-centre/kaiju.git 
 RUN cd /opt2/kaiju/src; make
+RUN cd /opt2 && git clone https://github.com/AstraZeneca-NGS/disambiguate.git
+RUN pip install pysam
+RUN chmod +x /opt2/disambiguate/disambiguate.py; ln -sf /opt2/disambiguate/disambiguate.py /usr/bin/disambiguate.py
```
Comment (skchronicles) [inline docker/NGS_QC/Dockerfile:13]: Is the plan to still use the biocontainer image for disambiguate?

**Review by skchronicles — APPROVED** [review]
If you can address some of the comments outlined within the next PR, that would be perfect!

File: docker/NGS_QC/Dockerfile (hunk 1 above)
Comment (rroutsong) [reply docker/NGS_QC/Dockerfile:13]: Yeah see: https://github.com/OpenOmics/weave/blob/340fc9a51bdc1b5eb1ab12b5093a85f4306bbda3/workflow/qc.smk#L92

**Review by skchronicles — COMMENTED** [review]
Thanks for the updates!

**Pull request merged**